Dockerfile

```diff
@@ -0,0 +1,51 @@
+FROM ubuntu:22.04
+
+RUN apt-get update && apt-get install -y \
+                    git \
+                    wget \
+                    flex \
+                    bison \
+                    gperf \
+                    python3 \
+                    python3-pip \
+                    python3-venv \
+                    python3-setuptools \
+                    cmake \
+                    ccache \
+                    libffi-dev \
+                    libssl-dev \
+                    dfu-util \
+                    libusb-1.0-0 \
+                    libgcrypt-dev \
+                    libglib2.0-dev \
+                    libfdt-dev \
+                    libpixman-1-dev \
+                    zlib1g-dev \
+                    ninja-build \
+                    libslirp-dev
+
+
+
+RUN mkdir -p ~/esp && cd ~/esp && git clone -b v4.4.7 --recursive https://github.com/espressif/esp-idf.git && cd esp-idf && ./install.sh && /bin/bash -c "source  ./export.sh"
```

Comment:
Also one more thing, when cloning, can you add specific git sha? To serve as dependency pinning.

Reply:
Wouldn't it be conflicting with `--depth 1` option? Also we use `-b v.4.4.7 `option to clone the specific tag.

Reply:
I think it should not, because `-b` show which git tree to copy, and `--depth` tells to avoid copying whole history. And these two should not be mutually exclusive. 

But we can try and test.

Reply:
oh, sha kind of adds a small amount of "cryptographic" safety, as we are pinning not the tag (which may be moved/changed), but actual contents (well hash of the contents) of the git tree. 

This can also be made to be a separate step (e.g. verifying the hash after clone), just to make sure that the dependency we downloaded is in-fact the one we are expecting to get.